src/components/contextMenu/AposContextMenu.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+  <div class="apos-context-menu" :class="modifiers">
+    <AposContextMenuTip />
+    <ul class="apos-context-menu__items">
+      <AposContextMenuItem 
+        v-for="item in menu" 
+        :key="item.action" 
+        :menuItem="item" 
+        v-on:clicked="itemClicked"
+      />
+    </ul>
+  </div>
+</template>
+
+<script>
+import AposLoading from '../loading/AposLoading.vue';
+import AposContextMenuItem from './AposContextMenuItem.vue';
+import AposContextMenuTip from './AposContextMenuTip.vue';
+
+export default {
+  name: 'AposContextMenu',
+  components: {
+    AposLoading,
+    AposContextMenuItem,
+    AposContextMenuTip
+  },
+  props: {
+    menu: Array,
+    tipAlignment: {
+      type: String,
+      default: 'left'
+    },
+    origin: {
+      type: String,
+      default: 'below'
+    },
+  },
+  computed: {
+    modifiers() {
+      let modifiers = 
+        `apos-context-menu--tip-alignment-${this.tipAlignment} 
+        apos-context-menu--origin-${this.origin}`;
+      return modifiers;
+    }
+  },
+  methods: {
+    itemClicked(action) {
+      console.log(`menu heard ${action} was clicked`);
+      this.$emit('itemClicked', action);
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+  .apos-context-menu {
+    position: relative;
+    display: inline-block;
+    margin: 10px;
+    &:before {
+      content: '';
+      width: 20px;
+      height: 20px;
+    }
+  }
+
+  .apos-context-menu--tip-alignment-left {
+    .apos-context-menu__tab {
+      left: 20px;
+    }
+  }
+
+  .apos-context-menu--tip-alignment-center {
+    .apos-context-menu__tab {
+      right: 0;
+      left: 0;
+      margin-left: auto;
+      margin-right: auto;
+    }
+  }
+  .apos-context-menu--tip-alignment-right {
+    .apos-context-menu__tab {
+      right: 20px;
+    }
+  }
+
+  .apos-context-menu--origin-below {
+    .apos-context-menu__tab {
+      top: -8.5px;
+    }
+  }
+
+  .apos-context-menu--origin-above {
+    .apos-context-menu__tab {
+      bottom: -9.5px;
```

Comment:
I'm getting this shadow line when the menu tip is below the thing. It looks like the bottom value has be slightly higher. `-9px` looks good.

<img width="299" alt="Screen Shot 2020-03-18 at 9 45 52 AM" src="https://user-images.githubusercontent.com/1155984/76972942-444af880-68fd-11ea-96e2-219d99efb870.png">

Reply:
take a look at the new commit, this might vary a hair by browser, i made the overlap in the svg's background color a bit longer to hopefully account for the half pixel difference